From 7c3c611d9c02fecf3fe37d6db8d96db93364efec Mon Sep 17 00:00:00 2001
From: Le Philousophe <lephilousophe@users.noreply.github.com>
Date: Thu, 24 Apr 2025 08:31:19 +0200
Subject: [PATCH 4/5] Update cctools-port to improve compatibility

Latest libraries from MacPorts need objc_msgSend support

Latest cctools-port need libdispatch
---
 build.sh | 19 +++++++++++++++++--
 1 file changed, 17 insertions(+), 2 deletions(-)

diff --git a/build.sh b/build.sh
index b87522b..4bb1f8c 100755
--- a/build.sh
+++ b/build.sh
@@ -145,10 +145,24 @@ if [ $NEED_TAPI_SUPPORT -eq 1 ]; then
   fi
 fi
 
+## Apple libdispatch Library ##
+
+get_sources https://github.com/tpoechtrager/apple-libdispatch.git $LIBDISPATCH_VERSION
+
+if [ $f_res -eq 1 ]; then
+  pushd $CURRENT_BUILD_PROJECT_NAME &>/dev/null
+  mkdir -p build
+  pushd build &>/dev/null
+  $CMAKE .. -DCMAKE_BUILD_TYPE=RELEASE -DCMAKE_INSTALL_PREFIX=$TARGET_DIR
+  make install -j$JOBS
+  popd &>/dev/null
+  build_success
+fi
+
 ## cctools and ld64 ##
 
-CCTOOLS_VERSION=986
-LINKER_VERSION=711
+CCTOOLS_VERSION=1024.3
+LINKER_VERSION=955.13
 
 get_sources \
   https://github.com/tpoechtrager/cctools-port.git \
@@ -162,6 +176,7 @@ if [ $f_res -eq 1 ]; then
   if [ $NEED_TAPI_SUPPORT -eq 1 ]; then
     CONFFLAGS+="--with-libtapi=$TARGET_DIR "
   fi
+  CONFFLAGS+="--with-libdispatch=$TARGET_DIR --with-libblocksruntime=$TARGET_DIR "
   CONFFLAGS+="--with-libxar=$TARGET_DIR "
   [ -n "$DISABLE_CLANG_AS" ] && CONFFLAGS+="--disable-clang-as "
   [ -n "$DISABLE_LTO_SUPPORT" ] && CONFFLAGS+="--disable-lto-support "
-- 
2.48.1

